woocommerce/woocommerce
# Handle experiment tests using transient-based api

### Description





This PR aims to fix an issue with object caching and testing experiments on the WCA Test Helper tool. 

**APIs introduced:**

- `(POST)/transients`
- `(GET)/transients`
- `(DELETE) /transients/<transient_names>`



### How to test the changes in this Pull Request:



Testing here is a bit tricky as it requires targeting an experiment. The easiest way is to create one with a snippet.

1. Use this tool on a site that uses object caching.
2. Target an existing experiment, or write a simple AB test snippet using the `\WooCommerce\Admin\Experimental_Abtest` class - check below.
3. Go to **Tools > WCA Test Helper > Experiments** and add the experiment name in the form. 
4. Refresh the page and notice that the experiment is not changing.
5. Apply this patch.
6. Go to **Tools > WCA Test Helper > Experiments** (do a hard refresh)
7. Toggle the experiment's status a few times until it's "treatment" again.
8. Return to the experiment page and notice that it correctly fetches the test value.

A very simple experiment snippet:
```php
add_action( 'init', function() {
    $in_treatment = \WooCommerce\Admin\Experimental_Abtest::in_treatment('my_test_experiment');
    echo $in_treatment ? 'TREATMENT' : 'CONTROL';
}
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
